### PR TITLE
feat(eventoverlay): add event path find for shadow elements

### PR DESCRIPTION
**Pull request by futurethang**
This change enables the Event Overlay component to have the correct `handleClickAway` behavior when used within an encapsulated Shadow DOM. Its normal mechanics check if `event.target` is contained within the anchor node, and closes the overlay if not. However, when used within a Shadow DOM, `event.target` gets reassigned to the wrapping web-component, and will continue all the way up the tree until it reaches the document level, and so always evaluates to an element that is outside of the anchor node.

!! ASSISTANCE NEEDED !!
This change introduces a utility function, and input is appreciated on the best implementation for the library's clean code. Input on whether/where else this function may be of use, build considerations for architecture-compliant compiling, or if it should be moved within the EventOverlay component itself.

## Description
This PR introduces a polyfill that checks the composed path of a clicked element more deeply, and resolves for cross browser compatibility. Within Event Overlay, this function is used to store an accurate reference to the original `event.target`, whether it lie inside a ShadowDOM or not. 

## Motivation and Context
This update is critical for the correct functionality for MFEs that utilize web components. There is not a solution that can be added to the project consuming the EventOverlay component.

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by futurethang** [comment]
I am thinking it may be better for the getEventPath function to simply live within the EventOverlay component. Unless we can identify 2 other areas it might come into play.

File: react/src/lib/EventOverlay/index.js [hunk 1]
```diff
@@ -224,69 +212,67 @@ class EventOverlay extends React.Component {
...
 
     anchorPosition.right =
       (rect.right || anchorPosition.left + anchorPosition.width) - parentRect.left;
     anchorPosition.bottom =
       (rect.bottom || anchorPosition.top + anchorPosition.height) - parentRect.top;
-    anchorPosition.middle =
-      anchorPosition.left + (anchorPosition.right - anchorPosition.left) / 2;
-    anchorPosition.center =
-      anchorPosition.top + (anchorPosition.bottom - anchorPosition.top) / 2;
+    anchorPosition.middle = anchorPosition.left + (anchorPosition.right - anchorPosition.left) / 2;
```
Comment (yananym) [inline react/src/lib/EventOverlay/index.js:222]: lots of formatting changes. might be an issue in general while working with Momentum - i've been scrolling for awhile and still haven't gotten to the change

**Comment by yananym** [comment]
circleci: test_www:
```
ERROR in ../react/es/EventOverlay/index.js
Module not found: Error: Can't resolve '../../helpers/getEventPath' in '/root/repo/react/es/EventOverlay'
 @ ../react/es/EventOverlay/index.js 13:0-51 297:24-33 312:24-33 320:24-33
```

**Comment by yananym** [comment]
test_core_percy:
```
@momentum-ui/core snapshot of activity-button:
     Uncaught Error: Cannot find module "../../helpers/getEventPath"
```

**Comment by yananym** [comment]
your relative path is fishy

**Comment by futurethang** [comment]
> your relative path is fishy

I think it is not necessary to modularize this function, I can put it right inside the component and avoid the unnecessary abstraction. Unless you are aware of other event.target related issues that happen in other components?

File: react/src/lib/EventOverlay/index.js [hunk 2]
```diff
@@ -315,53 +301,56 @@ class EventOverlay extends React.Component {
...
     const { close } = this.props;
 
     this.focusOnAnchorNode();
     close && close(e);
-  }
+  };
 
   handleCloseOnClick = e => {
     if (!this.props.isOpen) return;
+    const eventTarget = eventPath(e)[0];
```
Comment (futurethang) [inline react/src/lib/EventOverlay/index.js:357]: @yananym Look here for the applied changes (not just formatting), where `eventTarget` is a result of the polyfill, then checked in the `findDOMNode` below

**Comment by yananym** [comment]
@andrewholsted pleas have someone look at this - this is a pesky bug for when we use React components inside Shadow Dom and QA is going to be after us very soon if we don't merge the fix asap.

**Pull request merged**